test/ext/test_DataInterpolations.jl

```diff
@@ -0,0 +1,281 @@
+        @testset "input type: $(typeof(input))" for input in test_inputs
+            out_ref = t.interp(input)
+            s_ref = size(out_ref)
+
+            @testset "$T" for T in (DEFAULT_GRADIENT_TRACER, DEFAULT_HESSIAN_TRACER)
+                t_tracer = trace_input(T, input)
+                out_tracer = t.interp(t_tracer)
+                s_tracer = size(out_tracer)
+                @test s_tracer == s_ref
+            end
+            @testset "$T" for T in (
+                Dual{eltype(input),DEFAULT_GRADIENT_TRACER},
+                Dual{eltype(input),DEFAULT_HESSIAN_TRACER},
+            )
+                t_dual = trace_input(T, input)
+                out_dual = t.interp(t_dual)
+                s_dual = size(out_dual)
+                @test s_dual == s_ref
+                @test myprimal.(out_dual) ≈ out_ref
+            end
+        end
+    end
+end
+
+#===========#
+# Run tests #
+#===========#
+
+@testset "1D Interpolations" begin
+    @testset "$(testname(t))" for t in (
+        InterpolationTest(
+            1, ConstantInterpolation(u, t); is_der1_zero=true, is_der2_zero=true
+        ),
+        InterpolationTest(1, LinearInterpolation(u, t); is_der2_zero=true),
+        InterpolationTest(1, QuadraticInterpolation(u, t)),
+        InterpolationTest(1, LagrangeInterpolation(u, t)),
+        InterpolationTest(1, AkimaInterpolation(u, t)),
+        InterpolationTest(1, QuadraticSpline(u, t)),
+        InterpolationTest(1, CubicSpline(u, t)),
+        InterpolationTest(1, BSplineInterpolation(u, t, 3, :ArcLen, :Average)),
+        InterpolationTest(1, BSplineApprox(u, t, 3, 4, :ArcLen, :Average)),
+        # TODO: support when Julia 1.6 is dropped
+        # InterpolationTest(1, PCHIPInterpolation(u, t)), 
+        # InterpolationTest(1, CubicHermiteSpline(du, u, t)),
+        # InterpolationTest(1, QuinticHermiteSpline(ddu, du, u, t)),
+    )
+        test_jacobian(t)
+        test_hessian(t)
+        test_output(t)
+        yield()
+    end
+end
+
+for N in (2, 5)
+    local um = rand(N, length(t)) # matrix
+
+    @testset "$(N)D Interpolations" begin
+        @testset "$(testname(t))" for t in (
+            InterpolationTest(
+                N, ConstantInterpolation(um, t); is_der1_zero=true, is_der2_zero=true
+            ),
+            InterpolationTest(N, LinearInterpolation(um, t); is_der2_zero=true),
+            InterpolationTest(N, QuadraticInterpolation(um, t)),
+            InterpolationTest(N, LagrangeInterpolation(um, t)),
+            ## The following interpolations appear to not be supported on N dimensions as of DataInterpolations v6.2.0:
+            # InterpolationTest(N, AkimaInterpolation(um, t)),
+            # InterpolationTest(N, BSplineApprox(um, t, 3, 4, :ArcLen, :Average)),
+            # InterpolationTest(N, QuadraticSpline(um, t)),
+            # InterpolationTest(N, CubicSpline(um, t)),
+            # InterpolationTest(N, BSplineInterpolation(um, t, 3, :ArcLen, :Average)),
```

Comment:
And some methods only appear to support 1-dimensional states `u`.

Reply:
Yes, for some interpolation types there are different methods for different types of `u`, e.g.

```julia
_interpolate(A::LinearInterpolation{<:AbstractVector}, t::Number, iguess)
_interpolate(A::LinearInterpolation{<:AbstractMatrix}, t::Number, iguess)
```

If I'm not mistaken the only difference is the way `u` is indexed, not sure why that was never generalized